chainstate/src/detail/chainstateref.rs

```diff
@@ -266,23 +278,22 @@ impl<'a, S: BlockchainStorageRead, O: OrphanBlocks> ChainstateRef<'a, S, O> {
             std::cmp::Ordering::Equal => {}
         }
 
-        while first_block_index.block_id() != second_block_index.block_id()
-            && !first_block_index.is_genesis(self.chain_config)
-            && !second_block_index.is_genesis(self.chain_config)
-        {
-            first_block_index = self.get_previous_block_index(&first_block_index)?;
-            second_block_index = self.get_previous_block_index(&second_block_index)?;
+        loop {
+            match (&first_block_index, &second_block_index) {
```

Comment:
Isn't it better to be able to compare `Id<GenBlock>` and then that automatically figures out whether it's the same instead of handling cases here?

Reply:
Will have a look into this

Reply:
I think some matching is still necessary to get the parent (have to check whether genesis has been reached)